typescript/infra/src/roles.ts

```diff
@@ -15,4 +15,11 @@ export const ALL_KEY_ROLES = [
   Role.Kathy,
 ];
 
-export const ALL_AGENT_ROLES = [Role.Validator, Role.Relayer, Role.Scraper];
+// Use a const assertion to tell the compiler to retain the literal array item types.
+export const ALL_AGENT_ROLES = [
+  Role.Validator,
+  Role.Relayer,
+  Role.Scraper,
+] as const;
+export type ALL_AGENT_ROLES = typeof ALL_AGENT_ROLES[number];
```

Comment:
A little surprised they let you use the same name for a type and a value, wdyt of renaming the type to `AgentRole` or something like that to remove ambiguity?